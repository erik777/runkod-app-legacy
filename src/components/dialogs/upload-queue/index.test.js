import React from 'react';
import {DialogContent} from './index';
import TestRenderer from 'react-test-renderer';


const fnProps = {
  toggleUiProp: () => {
  },
  startUploadQueue: () => {
  },
  setUploadQueueConflictFlag: () => {
  },
  resetUploadQueue: () => {
  },
  fetchFiles: () => {
  }
};


test('1- uploading', () => {
  const props = Object.assign({}, {
    uploadQueue: {
      current: '/file1.json',
      inProgress: false,
      files: [
        {
          path: '/',
          name: 'file1.json'
        },
        {
          path: '/',
          name: 'file2.json'
        },
        {
          path: '/',
          name: 'file3.json'
        },
        {
          path: '/',
          name: 'file4.json'
        },
        {
          path: '/',
          name: 'file5.json'
        }
      ],

      completed: [],
      failed: [],
      skipped: [],
      conflict: false,
      conflictFlag: 0,
      log: []
    },
    ui: {
      uploadSummaryDetail: false
    },

  }, fnProps);

  const renderer = TestRenderer.create(
    <DialogContent {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});


test('2- uploading 2', () => {
  const props = Object.assign({}, {
    uploadQueue: {
      current: '/file3.json',
      inProgress: false,
      files: [
        {
          path: '/',
          name: 'file3.json'
        },
        {
          path: '/',
          name: 'file4.json'
        },
        {
          path: '/',
          name: 'file5.json'
        }
      ],

      completed: [
        {
          path: '/',
          name: 'file1.json'
        },
        {
          path: '/',
          name: 'file2.json'
        }
      ],
      failed: [],
      skipped: [],
      conflict: false,
      conflictFlag: 0,
      log: []
    },
    ui: {
      uploadSummaryDetail: false
    }
  }, fnProps);

  const renderer = TestRenderer.create(
    <DialogContent {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});


test('3- conflicted', () => {
  const props = Object.assign({}, {
    uploadQueue: {
      current: '/file3.json',
      inProgress: false,
      files: [
        {
          path: '/',
          name: 'file3.json'
        },
        {
          path: '/',
          name: 'file4.json'
        },
        {
          path: '/',
          name: 'file5.json'
        }
      ],

      completed: [
        {
          path: '/',
          name: 'file1.json'
        },
        {
          path: '/',
          name: 'file2.json'
        }
      ],
      failed: [],
      skipped: [],
      conflict: true,
      conflictFlag: 0,
      log: []
    },
    ui: {
      uploadSummaryDetail: false
    }
  }, fnProps);

  const renderer = TestRenderer.create(
    <DialogContent {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});


test('3- conflicted with single file', () => {
  const props = Object.assign({}, {
    uploadQueue: {
      current: '/file3.json',
      inProgress: false,
      files: [
        {
          path: '/',
          name: 'file3.json'
        }
      ],
      completed: [],
      failed: [],
      skipped: [],
      conflict: true,
      conflictFlag: 0,
      log: []
    },
    ui: {
      uploadSummaryDetail: false
    }
  }, fnProps);

  const renderer = TestRenderer.create(
    <DialogContent {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});


test('4- completed single file', () => {
  const props = Object.assign({}, {
    uploadQueue: {
      current: '',
      inProgress: false,
      files: [],
      completed: [
        {
          path: '/',
          name: 'file2.json'
        }
      ],
      failed: [],
      skipped: [],
      conflict: false,
      conflictFlag: 0,
      log: []
    },
    ui: {
      uploadSummaryDetail: false
    }
  }, fnProps);

  const renderer = TestRenderer.create(
    <DialogContent {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});


test('5- completed multi file', () => {
  const props = Object.assign({}, {
    uploadQueue: {
      current: '',
      inProgress: false,
      files: [],
      completed: [
        {
          path: '/',
          name: 'file2.json'
        },
        {
          path: '/',
          name: 'file3.json'
        }
      ],
      failed: [],
      skipped: [],
      conflict: false,
      conflictFlag: 0,
      log: []
    },
    ui: {
      uploadSummaryDetail: false
    }
  }, fnProps);

  const renderer = TestRenderer.create(
    <DialogContent {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});


test('6- completed mixed', () => {
  const props = Object.assign({}, {
    uploadQueue: {
      current: '',
      inProgress: false,
      files: [],
      completed: [
        {
          path: '/',
          name: 'file2.json'
        },
        {
          path: '/',
          name: 'file3.json'
        }
      ],
      failed: [
        {
          path: '/',
          name: 'file4.json'
        },
        {
          path: '/',
          name: 'file5.json'
        }
      ],
      skipped: [
        {
          path: '/',
          name: 'file6.json'
        }
      ],
      conflict: false,
      conflictFlag: 0,
      log: []
    },
    ui: {
      uploadSummaryDetail: false
    }
  }, fnProps);

  const renderer = TestRenderer.create(
    <DialogContent {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});


test('7- show details', () => {
  const props = Object.assign({}, {
    uploadQueue: {
      current: '',
      inProgress: false,
      files: [],
      completed: [
        {
          path: '/',
          name: 'file2.json'
        },
        {
          path: '/',
          name: 'file3.json'
        }
      ],
      failed: [
        {
          path: '/',
          name: 'file4.json'
        },
        {
          path: '/',
          name: 'file5.json'
        }
      ],
      skipped: [
        {
          path: '/',
          name: 'file6.json'
        }
      ],
      conflict: false,
      conflictFlag: 0,
      log: [
        {
          type: 'success',
          msg: 'file2.json uploaded'
        },
        {
          type: 'success',
          msg: 'file3.json uploaded'
        },
        {
          type: 'error',
          msg: 'file4.json could not uploaded'
        },
        {
          type: 'error',
          msg: 'file5.json could not uploaded'
        },
        {
          type: 'info',
          msg: 'file6.json skipped'
        }
      ]
    },
    ui: {
      uploadSummaryDetail: true
    }
  }, fnProps);

  const renderer = TestRenderer.create(
    <DialogContent {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});