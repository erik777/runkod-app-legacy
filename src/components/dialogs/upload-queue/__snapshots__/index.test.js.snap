// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- dialog visible but upload not started. 1`] = `
Array [
  <div
    className="progress-num"
  >
    0
     / 
    5
  </div>,
  <div
    className="progress-percent"
  >
    <div
      className="progress"
    >
      <div
        aria-valuemax={5}
        aria-valuemin={0}
        aria-valuenow={0}
        className="progress-bar progress-bar-animated progress-bar-striped"
        role="progressbar"
        style={
          Object {
            "width": "0%",
          }
        }
      />
    </div>
  </div>,
  <div
    className="upload-log-list"
  />,
]
`;

exports[`2- upload started. 1`] = `
Array [
  <div
    className="progress-num"
  >
    0
     / 
    5
  </div>,
  <div
    className="progress-percent"
  >
    <div
      className="progress"
    >
      <div
        aria-valuemax={5}
        aria-valuemin={0}
        aria-valuenow={0}
        className="progress-bar progress-bar-animated progress-bar-striped"
        role="progressbar"
        style={
          Object {
            "width": "0%",
          }
        }
      />
    </div>
  </div>,
  <div
    className="upload-log-list"
  />,
]
`;

exports[`3- 2 files uploaded. 1`] = `
Array [
  <div
    className="progress-num"
  >
    2
     / 
    5
  </div>,
  <div
    className="progress-percent"
  >
    <div
      className="progress"
    >
      <div
        aria-valuemax={5}
        aria-valuemin={0}
        aria-valuenow={2}
        className="progress-bar progress-bar-animated progress-bar-striped"
        role="progressbar"
        style={
          Object {
            "width": "40%",
          }
        }
      />
    </div>
  </div>,
  <div
    className="upload-log-list"
  >
    <div
      className="list-item list-item-success"
    >
      file2.json uploaded
    </div>
    <div
      className="list-item list-item-success"
    >
      file4.json uploaded
    </div>
  </div>,
]
`;

exports[`4- all files uploaded. 1`] = `
Array [
  <div
    className="progress-num"
  >
    3
     / 
    3
  </div>,
  <div
    className="progress-percent"
  >
    <div
      className="progress"
    >
      <div
        aria-valuemax={3}
        aria-valuemin={0}
        aria-valuenow={3}
        className="progress-bar progress-bar-animated progress-bar-striped"
        role="progressbar"
        style={
          Object {
            "width": "100%",
          }
        }
      />
    </div>
  </div>,
  <div
    className="upload-log-list"
  >
    <div
      className="list-item list-item-success"
    >
      file2.json uploaded
    </div>
    <div
      className="list-item list-item-success"
    >
      file4.json uploaded
    </div>
    <div
      className="list-item list-item-success"
    >
      file1.json uploaded
    </div>
    <div
      className="list-item list-item-info"
    >
      3 files uploaded ðŸŽ‰
    </div>
  </div>,
]
`;

exports[`5- finished with 1 fail. 1`] = `
Array [
  <div
    className="progress-num"
  >
    3
     / 
    3
  </div>,
  <div
    className="progress-percent"
  >
    <div
      className="progress"
    >
      <div
        aria-valuemax={3}
        aria-valuemin={0}
        aria-valuenow={3}
        className="progress-bar progress-bar-animated progress-bar-striped"
        role="progressbar"
        style={
          Object {
            "width": "100%",
          }
        }
      />
    </div>
  </div>,
  <div
    className="upload-log-list"
  >
    <div
      className="list-item list-item-success"
    >
      file2.json uploaded
    </div>
    <div
      className="list-item list-item-success"
    >
      file4.json uploaded
    </div>
    <div
      className="list-item list-item-success"
    >
      file1.json could not upload
    </div>
    <div
      className="list-item list-item-info"
    >
      2 files uploaded, 1 failed  ðŸ‘€
    </div>
  </div>,
  <div
    className="controls"
  >
    <button
      className="btn btn-outline-info btn-sm"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Close
    </button>
  </div>,
]
`;
